### DISCUSSION TITLE
I have some questions about 'Data Consistency'

### DISCUSSION THREAD

**Discussion by LiJie20190102** [JanusGraph/janusgraph#4347] [Q&A] [answered]
What does this line of code mean? I don't quite understand. I haven't tested any consistency check errors
![image](https://github.com/JanusGraph/janusgraph/assets/53458004/7c418651-efb9-49a2-b070-93b0992106ab)

`
  @Test
  public void testUnique(){
      JanusGraph graph = ConfiguredGraphFactory.open(GRAPH_NAME);

      JanusGraphManagement mgmt = graph.openManagement();
      PropertyKey name;
      if ((name = mgmt.getPropertyKey("consistentName")) == null) {
          name = mgmt.makePropertyKey("consistentName").dataType(String.class).make();
      }
      mgmt.setConsistency(name, ConsistencyModifier.LOCK); // Ensures only one name per vertex

      VertexLabel unique_test11 = mgmt.getVertexLabel("unique_test11");
      if (Objects.isNull(unique_test11)) {
          unique_test11 = mgmt.makeVertexLabel("unique_test11").make();
          mgmt.addProperties(unique_test11, name);
      }
    
      mgmt.commit();

      JanusGraphTransaction tx = graph.buildTransaction().checkExternalVertexExistence(false).start();
      tx.addVertex(T.id, "test1", T.label, "unique_test11", "consistentName", "aaa");
      tx.addVertex(T.id, "test2", T.label, "unique_test11", "consistentName", "aaa");
      tx.commit();
      
  }
`

**Comment by li-boxuan** [comment, marked as answer]
> mgmt.setConsistency(name, ConsistencyModifier.LOCK); // Ensures only one name per vertex

So that no two transactions can modify the name at the same time.
